### DISCUSSION TITLE
遇到了一个NPM安装问题

### DISCUSSION THREAD

**Discussion by jkiss** [ant-design/ant-design-mobile#4860] [Q&A] [answered]
如下图，ant-mobile的部分依赖装到了ant-mobile的包里面，导致代码找不到这些包。。。why？
![image](https://user-images.githubusercontent.com/3231326/155976667-7b3b70a4-310a-44a7-bb8b-fd4fd35e7e3b.png)

**Comment by awmleer** [comment, marked as answer]
这是预期行为啊……你说的“导致代码找不到这些包”具体是什么代码？

**Reply by jkiss** [reply]
就是 `ant-mobile` 找不到这些包。。。
![image](https://user-images.githubusercontent.com/3231326/156097739-b1473bc0-308b-4c5e-be96-2b2ea8a20572.png)

**Reply by awmleer** [reply]
你用的包管理器是？

**Reply by jkiss** [reply]
npm 8.3.1; node.js v16.14.0

**Reply by awmleer** [reply]
@jkiss 那打包工具呢？是 webpack 么？

**Reply by jkiss** [reply]
是的，5.69.1

**Reply by awmleer** [reply]
很奇怪，要不你把项目脱敏一下然后打个包上传上来看看？

**Reply by jkiss** [reply]
#### 我貌似发现问题了，如果我的项目同时安装 `antd`，那么就是下图的结果👇：
![截屏2022-03-02 上午11 04 10](https://user-images.githubusercontent.com/3231326/156288259-5643982f-4b51-4962-93c6-65d44e868491.png)

#### 如果没有安装 `antd`，就是下面这个了👇：
![截屏2022-03-02 上午11 05 26](https://user-images.githubusercontent.com/3231326/156288342-77df1de9-eaf4-45c2-85f4-3aeeb3f644d0.png)

### [这里是项目地址](https://github.com/jkiss/Ant-Mobile-Discussion-4860)

**Reply by awmleer** [reply]
复现出来了，只不过是 rc-field-form 被上提到项目根目录下的 node_modules 里了，这是正常的现象

**Reply by awmleer** [reply]
而且我 `npm run build` 是可以正常执行的，没有报错

**Reply by jkiss** [reply]
不好意思，我忘了引用 `antd-mobile` 了:sweat_smile:，又上传了一版：[repo](https://github.com/jkiss/Ant-Mobile-Discussion-4860)

**Reply by awmleer** [reply]
![image](https://user-images.githubusercontent.com/16526078/156492439-cc8bd49e-3060-41f1-a615-9dfa0d4d9d8f.png)

`webpack/resolve.js` 里，`modules` 加一下 `'node_modules'` 就可以了

**Reply by jkiss** [reply]
感谢大佬提供的解决方案～🙏
